extensions/notification/configManager/MongoConfigManager.js

```diff
@@ -91,49 +91,49 @@ class MongoConfigManager extends BaseConfigManager {
 
     /**
      * Connects to MongoDB using the MongoClientInterface
-     * and retreives the metastore collection
+     * and retrieves the metastore collection
      * @param {Function} cb callback
      * @returns {undefined}
      */
-    _setupMongoClient(cb) {
+    async _setupMongoClient(cb) {
         const mongoUrl = constructConnectionString(this._mongoConfig);
-        MongoClient.connect(mongoUrl, {
+        let client = new MongoClient(mongoUrl, {
             replicaSet: this._mongoConfig.replicaSet,
             useNewUrlParser: true,
-        },
-        (err, client) => {
-            if (err) {
-                this._logger.error('Could not connect to MongoDB', {
-                    method: 'MongoConfigManager._setupMongoClient',
-                    error: err.message,
-                });
-                return cb(err);
-            }
+            readPreference: this._mongoConfig.readPreference,
+        });
+
+        try {
+            client = await client.connect();
+
             this._logger.debug('Connected to MongoDB', {
                 method: 'MongoConfigManager._setupMongoClient',
             });
-            try {
-                this._mongoClient = client.db(this._mongoConfig.database, {
-                    ignoreUndefined: true,
-                });
-                this._metastore = this._mongoClient.collection(this._bucketMetastore);
+
+            this._mongoClient = client.db(this._mongoConfig.database, {
+                ignoreUndefined: true,
+            });
+            this._metastore = this._mongoClient.collection(this._bucketMetastore);
                 // get mongodb version
-                getMongoVersion(this._mongoClient, (err, version) => {
-                    if (err) {
-                        this._logger.error('Could not get MongoDB version', {
-                            method: 'MongoConfigManager._setupMongoClient',
-                            error: err.message,
-                        });
-                        return cb(err);
-                    }
-                    this._mongoVersion = version;
-                    return cb();
-                });
-                return undefined;
-            } catch (error) {
-                return cb(error);
-            }
-        });
+            getMongoVersion(this._mongoClient, (err, version) => {
+                if (err) {
+                    this._logger.error('Could not get MongoDB version', {
+                        method: 'MongoConfigManager._setupMongoClient',
+                        error: err.message,
+                    });
+                    return cb(err);
+                }
+                this._mongoVersion = version;
+                return cb();
+            });
+            return undefined;
```

Comment:
merge the return statement and invocation of the last "callback-passing" function : to make it more explicit that the processing goes on asynchronously
```suggestion
            return getMongoVersion(...);
```